src/officehours_api/tests.py

```diff
@@ -163,3 +165,78 @@ def test_meeting_start_notifies_attendees(self, mock_twilio: mock.MagicMock):
             receivers >=
             {'+15555550000', '+15555550001'}
         )
+
+
+class MeetingSerializerTestCase(TestCase):
+    """
+    Test the scenarios where the backend type is invalid, valid,
+    patched to none, and patched to invalid.
+    """
+
+    def setUp(self):
+        self.user1 = User.objects.create(username='user1')
+        self.user2 = User.objects.create(username='user2')
+        self.inperson_queue = Queue.objects.create(name='test queue', status='open', allowed_backends=['inperson'])
+        self.zoom_queue = Queue.objects.create(name='test queue', status='open', allowed_backends=['zoom'])
+        self.inperson_queue.hosts.add(self.user1)
+        self.zoom_queue.hosts.add(self.user1)
+
+    def test_backend_type_invalid(self):
+        data = {
+            'queue': self.zoom_queue.id,
+            'attendee_ids': [self.user2.id],
+            'agenda': 'test agenda',
+            'assignee_id': self.user1.id,
+            'backend_type': 'inperson'
+        }
+        serializer = MeetingSerializer(data=data)
+        with self.assertRaises(ValidationError) as cm:
+            serializer.is_valid(raise_exception=True)
+        error = str(cm.exception.detail['non_field_errors'][0])
+        self.assertEqual(error, "inperson is not one of the queue's allowed backend types (['zoom'])")
+
+    def test_backend_type_valid(self):
+        data = {
+            'queue': self.inperson_queue.id,
+            'attendee_ids': [self.user2.id],
+            'agenda': 'test agenda',
+            'assignee_id': self.user1.id,
+            'backend_type': 'inperson'
+        }
+        serializer = MeetingSerializer(data=data)
+        valid = serializer.is_valid(raise_exception=False)
+        self.assertTrue(valid)
+
+    def test_patch_backend_type_none(self):
+        meeting = Meeting(**{
+            'queue': self.inperson_queue,
+            'agenda': 'test agenda',
+            'assignee_id': self.user1.id,
+            'backend_type': 'inperson'
+        })
+        # Patch the meeting without a backend type
+        data = {
+            'attendee_ids': [self.user2.id],
+        }
```

Comment:
Sorry about all the back and forth. Can you specify the attendee in the setup in both these cases? Then maybe just patch the `agenda` in this one, and then the `backend_type` in the other? In the UI, the attendee is basically always set on the initial meeting creation and then never changed. We may as well try to make these realistic or follow real-world usage.

Reply:
Of course! I thought we cannot assign attendees when initialization because when I tried to include `attendee_ids` it would throw an error `Meeting() got an unexpected keyword argument 'attendee_ids'`. But then I realized I forgot to reference the definition of Meeting() in models.py and I just need to include `attendee` from the setup. So now it would work I think!

Reply:
You're right, I wasn't thinking this through completely. Also apparently I'm a little rusty with the Django ORM. What your doing isn't throwing an error, but it's also not creating an attendee. We have to use special methods to establish this relationship with the other models (`Attendee`, `User`) since it's a [Many-to-Many Relationship](https://docs.djangoproject.com/en/4.2/topics/db/examples/many_to_many/).

I think this is probably the appropriate way to create attendees.

```
        meeting = Meeting(**{
            'queue': self.inperson_queue,
            'agenda': 'test agenda',
            'assignee': self.user1,
            'backend_type': 'inperson'
        })
        meeting.save()
        meeting.attendees.add(self.user2)
        meeting.save()
        logger.info(meeting)
        logger.info(meeting.attendees.all())
```

When debugging this, I recommend adding a logger to the file using `logging.getLogger` and then using `logger.info` to look at the state of the `Meeting` or attendees, like I am above.

This doesn't really affect your tests, but we may as well get the attendee adding right. Could you update these cases one last time please?

Reply:
Of course, should I define the logger object `logger = logging.getLogger(__name__)` at the beginning of this file or just within those two test case functions?

Reply:
Beginning of the file. You don't have to commit that though, it's just for your debugging/exploration.

Reply:
Noted! So I just need to change them into this and it should be fine for commit right?
```
meeting = Meeting(**{
            'queue': self.inperson_queue,
            'agenda': 'test agenda',
            'assignee': self.user1,
            'backend_type': 'inperson'
        })
        meeting.save()
        meeting.attendees.add(self.user2)
        meeting.save()
```

Reply:
Yep, looks good.

Reply:
I have pushed the latest commit([52d435a](https://github.com/tl-its-umich-edu/remote-office-hours-queue/pull/400/commits/52d435af99e863f2804a646e6d8627cee024ac23)) to resolve this.